src/storage/src/mut_records.rs

```diff
@@ -24,32 +32,32 @@ use crate::records::FileRecords;
 
 pub const MESSAGE_LOG_EXTENSION: &str = "log";
 
-/// MutFileRecordsFlushPolicy describes and implements a flush policy
-pub enum MutFileRecordsFlushPolicy {
-    NoFlush,
-    EveryWrite,
-    CountWrites { n_writes: u32, write_tracking: u32 },
-}
-
 /// Can append new batch to file
 pub struct MutFileRecords {
     base_offset: Offset,
     item_last_offset_delta: Size,
-    f_sink: BoundedFileSink,
-    flush_policy: MutFileRecordsFlushPolicy,
+    f_sink: Arc<Mutex<BoundedFileSink>>,
```

Comment:
Since underlying File system is thread safe.  Is it better to create separate file from descriptor rather than tryin to use mutex?

Reply:
The file descriptor isn't accessible from the boundedfilesink / or the async file type, so there's a tradeoff of changing that and maintining the difference or upstreaming the change.  

You're right about the fs side of things supporting thread safe part, but copying fs descriptor sort of gives added cleanup issues on the fd on drop of the parent BoundedFileSink. that code needs to then not close the fd and the flush task needs to know to close it.  I suspect this would be easier to address  one of the io_uring backends, and maybe defer improving this beyond a Mutex for now?

Reply:
sounds goo